api/core/model_runtime/model_providers/__base/tokenizers/gpt2_tokenzier.py

```diff
@@ -20,7 +23,8 @@ def _get_num_tokens_by_gpt2(text: str) -> int:
 
     @staticmethod
     def get_num_tokens(text: str) -> int:
-        return GPT2Tokenizer._get_num_tokens_by_gpt2(text)
+        future = _executor.submit(GPT2Tokenizer._get_num_tokens_by_gpt2, text)
+        return future.result()
```

Comment:
It sounds like still a synchronized way. `future.result()` is only returned until the result are waited.

Reply:
You can give it a try with Gevent.

Reply:
I'm not sure how much the gevent helps in this case.
As in Python's doc, `.result()` is claimed to wait until the future instance completed.
https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.wait